core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/FlowControl.scala

```diff
@@ -99,8 +104,15 @@ class FlowControl {
         Constants.unprocessedBatchesCreditLimitPerSender
       ) > 0
     ) {
-      messageBuffer.append(dataMessagesAwaitingCredits(receiverId).dequeue())
-      receiverIdToCredits(receiverId) = receiverIdToCredits(receiverId) - 1
+      val msg = dataMessagesAwaitingCredits(receiverId).dequeue()
+      messageBuffer.append(msg)
+      msg match {
+        case dataMsg: WorkflowDataMessage =>
+          receiverIdToCredits(receiverId) =
+            receiverIdToCredits(receiverId) - getInMemSize(dataMsg).intValue()
+        case _ =>
+          receiverIdToCredits(receiverId) = receiverIdToCredits(receiverId) - 1
```

Comment:
did we decide not to get the size of control messages?

Reply:
if we decide not to do an actual size for the control message, it might be better to use a rough number. using 1 sounds too small.